university/src/university/StudentFactory.java

```diff
@@ -0,0 +1,14 @@
+package university;
+
+public class StudentFactory implements AbstractFactory<Student> {
+
+    @Override
+    public Student createWithStringInput(String studentType, String name, String studentId, Faculty major) {
+        if (studentType.equalsIgnoreCase("BScStudent")) {
+            return new BScStudent(name, studentId, major);
+        } else if (studentType.equalsIgnoreCase("MScStudent")) {
```

Comment:
student types can be strings, but can be enums too - so you could skip line 12 and you wouldn't let anyone pass invalid student types